pySDC/projects/PinTSimE/battery_2capacitors_model.py

```diff
@@ -0,0 +1,304 @@
+import numpy as np
+import dill
+from pathlib import Path
+
+from pySDC.helpers.stats_helper import get_sorted
+from pySDC.core.Collocation import CollBase as Collocation
+from pySDC.implementations.problem_classes.Battery import battery_n_capacitors
+from pySDC.implementations.sweeper_classes.imex_1st_order import imex_1st_order
+from pySDC.implementations.controller_classes.controller_nonMPI import controller_nonMPI
+from pySDC.projects.PinTSimE.battery_model import get_recomputed, proof_assertions_description
+from pySDC.projects.PinTSimE.piline_model import setup_mpl
+import pySDC.helpers.plot_helper as plt_helper
+from pySDC.core.Hooks import hooks
+
+from pySDC.projects.PinTSimE.switch_estimator import SwitchEstimator
+
+
+class log_data(hooks):
```

Comment:
Maybe one hook for the whole project is enough? You log really similar things most of the time. If the solution is different, you can just log the whole solution and separate it into components later.

I recently tried to streamline this a bit, so, if you want to keep the names in the stats object, you can also have one hook that logs all quantities that are the same and then make separate hooks that log things that may be different across problems. Then you just add multiple hooks as a list.

Reply:
I saw that you used type='u' for the whole solution u? So, it is possible to do this in my hooks? In case of yes I can adapt my hooks, and for each number of capacitors I can use only one hook I guess.

Reply:
Yes, you can do it too!
In your hook currently you add `L.uend[1]` as `voltage C` for instance.
If you add `L.uend` as `u`, for instance by using the [predefined hook](https://github.com/Parallel-in-Time/pySDC/blob/53d6faf64827db05d6274f44002e0ab3f44a4b9d/pySDC/implementations/hooks/log_solution.py#L4), you can sort everything out like this:
```
u = get_sorted(stats, type='u')
V_C = [me[1][1] for me in u]
```
You can just add the index back when sorting the stats if you leave it out when logging them.

I started the way you did it because it's nice to have sensible names for everything, but for the sake of abstraction, I just went ahead and called everything `u` :D